pkg/lib/_internal/dnf5daemon.ts

```diff
@@ -297,28 +340,299 @@ export class Dnf5DaemonManager implements PackageManager {
                 });
         }
 
-        const client = dbus_client();
-        const subscription = client.subscribe({}, signal_emitted);
-        let session: string | null = null;
+        await this.with_session(async (session) => {
+            try {
+                await call(session, "org.rpm.dnf.v0.rpm.Rpm", "install", [data.missing_names, {}]);
+                const [, resolve_result] = await call(session, "org.rpm.dnf.v0.Goal", "resolve", [{}]);
 
-        try {
-            session = await open_session();
-            await call(session, "org.rpm.dnf.v0.rpm.Rpm", "install", [data.missing_names, {}]);
-            const [, resolve_result] = await call(session, "org.rpm.dnf.v0.Goal", "resolve", [{}]);
+                if (resolve_result !== 0) {
+                    const [problem] = await call(session, "org.rpm.dnf.v0.Goal", "get_transaction_problems_string", []);
+                    throw new ResolveError(`Resolving install failed with result=${resolve_result} ${problem}`);
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22611-75a4849c-20251127-151944-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)